index.js

```diff
@@ -0,0 +1,184 @@
+				airports = body;
+				return this.getIATA(airports, search);
+			}).catch((error) => {
+				logger.error("Caught error:", error);
+			})
+		}
+	}
+ 	/**
+   * Process a airports JSON to an options list
+   * @param {*} airports
+   * @param {String} search 
+   */
+	async getIATA(airports, search) {
+		if (search != null) {
+		  search = search.trim().toLowerCase();
+		}
+		let option = [];
+		for (const airport of airports){
+			const key = airport.iata;
+			let value = airport.name || '';
+			if (!search || value.toLowerCase().includes(search)) {
+				option.push({
+					key: key,
+					value: value
+				});
+			}	
+		}
+		return option;
+	}	
+
+	// Store price obtained from last update
+	setLastPrice(price) {
+		this.store.put("lastPrice", price);
+		return true;
+	}
+
+	// Retrieve stored price data from last update
+	getLastPrice() {
+		if (this.store.get("lastPrice") != null) {
+			return this.store.get("lastPrice");
+		} else {
+			// If no price is stored
+			return null;
+		}
+	}
+
+	async run() {
+		// Compare the current price to the old price
+		return this.getPrice().then(new_price => {
+			const old_price = this.getLastPrice();
+			logger.info(`The new price is ${new_price}`);
+			logger.info(`The old price is ${old_price}`);
+			let color;
+			let message;
+			// If there is no price stored in localStorage
+			if (old_price == null) {
+				if (new_price == null) {
+					color = '#DF0101'; // red
+					message = `This flight is not listed. Please modify your request.`;
+				} else if (new_price != null) {
+					color = '#088A08'; // green
+					message = `The best price for this flight is ${new_price} ${this.config.currency}.`;
+				}
+			} else if (new_price <= old_price) {
+				if (new_price == null) {
+					color = '#DF0101'; // red
+					message = `This flight is not listed. Please modify your request.`;
+				} else {
+					color = '#088A08'; // green
+					message = `The best price for this flight is ${new_price} ${this.config.currency}.
+					Let's buy it!`;
+				}
+			} else if (new_price > old_price) {
+				if (new_price == null) {
+					color = '#DF0101'; // red
+					message = `This flight is not listed. Please modify your request.`;
+				} else {
+					color = '#DF0101'; // red
+					message = `The best price was ${old_price} ${this.config.currency} 
+					and is now ${new_price} ${this.config.currency}`;
+				}
+			}
+			const a = new q.Signal({
+				points: [
+					[new q.Point(color)]
+				],
+				name: 'FlightPriceWatcher',
+				message: message,
+				errors: 'Error signal'
+			});
+			this.setLastPrice(new_price);
+			return a;
+		})
+	}
+
+	formatDate() {
+		const departDate = this.config.departDate;
+		const returnDate = this.config.returnDate;
+		var regEx = /^\d{4}-\d{2}-\d{2}$/;	
+		if((!departDate || departDate.match(regEx)) && (!returnDate || returnDate.match(regEx))) {
+			return true; //Valid format
+		} else {
+			throw new Error("Error validating format");  // Invalid format
+		}
+	}
```

Comment:
This function is not to formatDate. This is to check the format of the date . Should change the name of it. 
Also it should also return a boolean. Here you are throwing an error if not valid format